api/transport/router.go

```diff
@@ -37,6 +37,21 @@ type Procedure struct {
 
 	// HandlerSpec specifiying which handler and rpc type.
 	HandlerSpec HandlerSpec
+
+	// Encoding of the handler, for introspection.
+	Encoding Encoding
+
+	// Signature of the handler, for introspection. This should be a snippet of
+	// Go code representing the function definition.
+	Signature string
+}
+
+// Less returns true if a.(Service, Name) < b.(Service, Name).
+func (a Procedure) Less(b Procedure) bool {
```

Comment:
Would be useful to described *why* this function needs to exist, or *how* it's used.